src/applications/hca/containers/IDPage.jsx

```diff
@@ -16,10 +19,19 @@ class IDPage extends React.Component {
     focusElement('.va-nav-breadcrumbs-list');
   }
 
-  render() {
+  componentDidUpdate(prevProps) {
```

Comment:
Is there a better way to gate this page/route based on if the user is logged in or not? This totally works, but feels a little gross.

As an aside, it looks like there's no way to change the post-login redirect. The user is simply redirected back to the route they were on when they logged in and there's no way to change that?

Reply:
The only better way I can think of is to have the reducer for this app hook into the action that updates the logged in state and explicitly add something to the state to tell the component to navigate (rather than implicitly relying on the `isLoggedIn` state). 

We could also navigate to the destination before redirecting to the login. Wouldn’t need to render anything- just push into that route and use state to tell it it needs to redirect (not sure where this should live- action creator / component wise). We can assume that the login will succeed and optimize for that flow and have the destination redirect to an error state if the login fails?

Reply:
These are good suggestions. I hadn't even thought of adding another reducer to act on the `isLoggedIn` state; it hadn't even occurred to me that that was possible. Duh. 🙄 

I'm going to leave things as is for now because I think that hiding the reason for why a redirect would happen would make this component a little harder to understand. "Oh, the `shouldShowIDPage` state is `false` so this will redirect. But why is `shouldShowIDPage` set to `false` in the first place?"

Navigating to the destination _before_ showing the login page is a cool idea. But it won't deal with gating this page if someone navigates directly to it. That's a tiny edge case, but still possible